flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -198,21 +199,28 @@ private void startAsynchronouslySplit() {
                 this.executor = Executors.newSingleThreadExecutor(threadFactory);
             }
 
-            executor.submit(
-                    () -> {
-                        Iterator<TableId> iterator = remainingTables.iterator();
-                        while (iterator.hasNext()) {
-                            TableId nextTable = iterator.next();
-                            // split the given table into chunks (snapshot splits)
-                            Collection<MySqlSnapshotSplit> splits =
-                                    chunkSplitter.generateSplits(nextTable);
-                            synchronized (lock) {
-                                remainingSplits.addAll(splits);
-                                remainingTables.remove(nextTable);
-                                lock.notify();
-                            }
-                        }
-                    });
+            Future<?> future =
+                    executor.submit(
+                            () -> {
+                                Iterator<TableId> iterator = remainingTables.iterator();
+                                while (iterator.hasNext()) {
+                                    TableId nextTable = iterator.next();
+                                    // split the given table into chunks (snapshot splits)
+                                    Collection<MySqlSnapshotSplit> splits =
+                                            chunkSplitter.generateSplits(nextTable);
+                                    synchronized (lock) {
+                                        remainingSplits.addAll(splits);
+                                        remainingTables.remove(nextTable);
+                                        lock.notify();
+                                    }
+                                }
+                            });
+
+            try {
+                future.get();
```

Comment:
The chunk splitting logic is designed to run asynchronously as an optimization. Waiting for the future to finish here will break the async mechanism.